From a372906a52fe874c261bc64411ba3f5aaf2978d5 Mon Sep 17 00:00:00 2001
From: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
Date: Tue, 10 Nov 2009 02:41:56 -0800
Subject: [PATCH 2/4] MIPS: nec_emma3pf: emma3pf.h: Replace EMMA_IRQ_INTn with EMMA_IRQ_INT(n)

We don't want to see a lot of unused / useless EMMA_IRQ_INTn defines.
EMMA_SW_IRQ_INTn should be fixed as well.

Signed-off-by: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
---
 include/asm-mips/emma/emma3pf.h |  143 ++++----------------------------------
 1 files changed, 15 insertions(+), 128 deletions(-)

diff --git a/include/asm-mips/emma/emma3pf.h b/include/asm-mips/emma/emma3pf.h
index 9e48d0b..13c6f40 100644
--- a/include/asm-mips/emma/emma3pf.h
+++ b/include/asm-mips/emma/emma3pf.h
@@ -146,6 +146,9 @@
 #define EMMA_PCI_CONFIG_BASE	EMMA_PCI_IO_BASE
 #define EMMA_PCI_CONFIG_SIZE	EMMA_PCI_IO_SIZE
 
+/*
+ * IRQ defines
+ */
 #define NUM_EMMA_IRQ		96
 #define NUM_EMMA_IRQ_SW		32
 #define NUM_EMMA_IRQ_GPIO	57
@@ -154,135 +157,19 @@
 #define EMMA_SW_IRQ_BASE	(EMMA_IRQ_BASE + NUM_EMMA_IRQ)
 #define EMMA_GPIO_IRQ_BASE	(EMMA_SW_IRQ_BASE + NUM_EMMA_IRQ_SW)
 
-/*
- * emma3pf irq defs
- */
-
-#define EMMA_IRQ_INT00		(0 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT01		(1 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT02		(2 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT03		(3 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT04		(4 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT05		(5 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT06		(6 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT07		(7 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT08		(8 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT09		(9 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT10		(10 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT11		(11 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT12		(12 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT13		(13 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT14		(14 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT15		(15 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT16		(16 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT17		(17 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT18		(18 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT19		(19 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT20		(20 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT21		(21 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT22		(22 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT23		(23 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT24		(24 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT25		(25 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT26		(26 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT27		(27 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT28		(28 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT29		(29 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT30		(30 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT31		(31 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT32		(32 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT33		(33 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT34		(34 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT35		(35 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT36		(36 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT37		(37 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT38		(38 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT39		(39 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT40		(40 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT41		(41 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT42		(42 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT43		(43 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT44		(44 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT45		(45 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT46		(46 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT47		(47 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT48		(48 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT49		(49 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT50		(50 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT51		(51 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT52		(52 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT53		(53 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT54		(54 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT55		(55 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT56		(56 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT57		(57 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT58		(58 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT59		(59 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT60		(60 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT61		(61 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT62		(62 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT63		(63 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT64		(64 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT65		(65 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT66		(66 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT67		(67 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT68		(68 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT69		(69 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT70		(70 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT71		(71 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT72		(72 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT73		(73 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT74		(74 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT75		(75 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT76		(76 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT77		(77 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT78		(78 + EMMA_IRQ_BASE)
-#define EMMA_IRQ_INT79		(79 + EMMA_IRQ_BASE)
-
-#define EMMA_SW_IRQ_INT00	(0 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT01	(1 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT02	(2 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT03	(3 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT04	(4 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT05	(5 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT06	(6 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT07	(7 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT08	(8 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT09	(9 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT10	(10 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT11	(11 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT12	(12 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT13	(13 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT14	(14 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT15	(15 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT16	(16 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT17	(17 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT18	(18 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT19	(19 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT20	(20 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT21	(21 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT22	(22 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT23	(23 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT24	(24 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT25	(25 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT26	(26 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT27	(27 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT28	(28 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT29	(29 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT30	(30 + EMMA_SW_IRQ_BASE)
-#define EMMA_SW_IRQ_INT31	(31 + EMMA_SW_IRQ_BASE)
-
-#define EMMA_SW_CASCADE		EMMA_IRQ_INT06
-#define EMMA_GPIO_CASCADE	EMMA_IRQ_INT46
+#define EMMA_IRQ_INT(n)		(EMMA_IRQ_BASE + (n))
+#define EMMA_SW_IRQ_INT(n)	(EMMA_SW_IRQ_BASE + (n))
 
-#define EMMA_IRQ_PUART0		EMMA_IRQ_INT48
-#define EMMA_IRQ_PUART1		EMMA_IRQ_INT49
-#define EMMA_IRQ_PIIC0		EMMA_IRQ_INT51
-#define EMMA_IRQ_PIIC1		EMMA_IRQ_INT52
-#define EMMA_IRQ_PIIC2		EMMA_IRQ_INT53
+#define EMMA_SW_CASCADE		EMMA_IRQ_INT(6)
+#define EMMA_GPIO_CASCADE	EMMA_IRQ_INT(46)
 
-#define EMMA_IRQ_USB		EMMA_IRQ_INT35
-#define EMMA_IRQ_KSATA0		EMMA_IRQ_INT36
-#define EMMA_IRQ_ETHER		EMMA_IRQ_INT61
+#define EMMA_IRQ_USB		EMMA_IRQ_INT(35)
+#define EMMA_IRQ_KSATA0		EMMA_IRQ_INT(36)
+#define EMMA_IRQ_PUART0		EMMA_IRQ_INT(48)
+#define EMMA_IRQ_PUART1		EMMA_IRQ_INT(49)
+#define EMMA_IRQ_PIIC0		EMMA_IRQ_INT(51)
+#define EMMA_IRQ_PIIC1		EMMA_IRQ_INT(52)
+#define EMMA_IRQ_PIIC2		EMMA_IRQ_INT(53)
+#define EMMA_IRQ_ETHER		EMMA_IRQ_INT(61)
 
 #endif	/* __ASM_EMMA_EMMA3PF_H */
-- 
1.6.5.2

